### PR TITLE
♿️  #456: improve modals

### PR DESCRIPTION
**Pull request by samrichca**
### Consent UI
- Add react-modal library
- Add modal for date of birth field on the register page
  - Open modal on blur if the user is a minor
- Refactor modals on invite and consent pages to use the new modal
- Improve CSS for modals on mobile, and modals with long content

## Instructions

Run `pnpm i` to add react-modal

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/components/common/Modal/Modal.module.scss [hunk 1]
```diff
@@ -1,28 +1,46 @@
-.card {
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/Modal/Modal.module.scss:1]: kept the CSS the same, just reorganized it into 1 file

File: apps/consent-ui/src/components/views/Register/RegisterForm/FormStep2.tsx [hunk 2]
```diff
@@ -103,16 +100,7 @@ const FormStep2 = ({
 
 		if (recaptchaToken) {
 			const data = Object.assign({}, step1Data, step2Data);
-			axiosClient
```
Comment (samrichca) [inline apps/consent-ui/src/components/views/Register/RegisterForm/FormStep2.tsx:106]: remove client-side axios request due to page crashes

File: apps/consent-ui/src/components/common/Form/fieldsets/inputs/CalendarInput.tsx [hunk 3]
```diff
@@ -52,21 +54,27 @@ const CalendarInput = <T extends FieldValues>({
...
-					onBlur={onBlur}
-					disabled={disabled}
-					{...ariaProps}
-					aria-required={required}
-				/>
-			)}
+			render={({ field: { onBlur: controlOnBlur, onChange, value } }) => {
+				const handleBlur = () => {
+					controlOnBlur();
+					onBlur();
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/Form/fieldsets/inputs/CalendarInput.tsx:60]: passing down onBlur so I can trigger the modal with it

**Review by mistryrn — COMMENTED** [review]
New modal looks and works great! Thanks for finding this library and updating all the modals to use it.

Only thing I noticed was this warning in the console:
Screenshot 2024-03-07 at 9 57 18 AM
```
Warning: react-modal: App element is not defined. Please use `Modal.setAppElement(el)` or set `appElement={el}`. This is needed so screen readers don't see main content when modal is opened. It is not recommended, but you can opt-out by setting `ariaHideApp={false}`.
```

Do you mind taking a look at that before we merge this?

File: apps/consent-ui/src/components/common/PageLayout/index.tsx [hunk 4]
```diff
@@ -34,15 +33,13 @@ const PageLayout = async ({
...
-			<NotificationProvider>
-				<Container>
-					<Header currentLang={currentLang} />
-					<main className={styles.main}>{children}</main>
-					<Footer currentLang={currentLang} />
-				</Container>
-			</NotificationProvider>
-		</ModalProvider>
+		<NotificationProvider>
+			<Container id="root">
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/PageLayout/index.tsx:37]: next removed the ID on the topmost div in version 13 so I added one here to support the modal's a11y functions

**Review by mistryrn — APPROVED** [review]
Thanks for fixing that warning! 🚀

**Pull request merged**